src/Data/Graph.purs

```diff
@@ -30,6 +30,11 @@ newtype Graph k v = Graph (Map k (Tuple v (List k)))
 instance functorGraph :: Functor (Graph k) where
   map f (Graph m) = Graph (map (lmap f) m)
 
+instance foldableGraph :: Foldable (Graph k) where
+  foldl   f z (Graph m) = foldl   f z $ fst <$> M.values m
+  foldr   f z (Graph m) = foldr   f z $ fst <$> M.values m
+  foldMap f   (Graph m) = foldMap f   $ fst <$> M.values m
```

Comment:
Ideally, we would use 1 fold rather than 2 here (i.e. `M.values` folds once, then `<$>` folds a second time). But the single fold version...
`foldl f z (foldl (\acc (Tuple k _) -> Array.snoc acc k) [] m)`
... is likely less performant because of the usage of `Array.snoc`.

Reply:
Would something like this be better?
```haskell
instance foldableGraph :: Foldable (Graph k) where
  foldl   f z (Graph m) = foldl   (\acc (Tuple k _) -> f acc k) z $ M.values m
  foldr   f z (Graph m) = foldr   (\(Tuple k _) acc -> f k acc) z $ M.values m
  foldMap f   (Graph m) = foldMap (f <<< fst) $ M.values m
```

Reply:
Ah... probably.

Reply:
Done in 20b2649